packages/at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -154,7 +154,7 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
       final serverCommitId = notification.value;
       int localCommitId = -1;
       try {
-        localCommitId = await _getLocalCommitId();
+        localCommitId = await getLastReceivedServerCommitId();
```

Comment:
Please add an explanation of this change to the PR description. Is there a test (of any kind - unit test, functional test or end-to-end test) which checks that this code is working correctly?

Reply:
Thanks for updating the description. Is there a test which will fail if this change is reverted?

Reply:
> Thanks for updating the description. Is there a test which will fail if this change is reverted?

@sitaram-kalluri ^^

Reply:
@gkc : Yes added a unit test too in the followng commit. https://github.com/atsign-foundation/at_client_sdk/pull/1073/commits/24100f61e445db64f7e361a664e3ee91e00c4dff#diff-cc97e657d252f9b053e893ca0682e931ff543e6aa9969febb9a7e6e75321955f